.github/workflows/plan.yml

```diff
@@ -2,24 +2,35 @@ name: terraform
 on:
   pull_request:
     branches: [main]
+  pull_request_target:
+    branches: [main]
 
 jobs:
   plan:
-    runs-on: ubuntu-latest
     env:
+      DEPLOYERS: |
+          [
+            "konturn@wustl.edu"
+          ]
       AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
       AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
       GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    runs-on: ubuntu-latest
+    #These steps run if either the PR is within the same repo or if the PR is on a fork and the committer has deployer access
     steps:
       - uses: actions/checkout@v2
+        if: ${{ ((github.event.pull_request.head.repo.full_name != github.repository) && contains( env.DEPLOYERS, github.event.pusher.email ) && github.event_name == 'pull_request_target') || (github.event.pull_request.head.repo.full_name == github.repository && github.event_name == 'pull_request') }}
```

Comment:
What happens if I just change my email address in my commit to one in this list? Is there a way to check if they submitted the change with a valid GPG key?

Reply:
Yeah there's an action I found which makes sure a commit is verified--I'll add it in.

Reply:
Due to limitations in said action, ended up just using Python's Github libraries directly to query the verification status of the latest commit on the PR branch. I also made some other overhauls that simplifies the syntax a bit.